update and add files to compile Matlab interface on Windows

lenvm opened the pull request:
Update:
- README.md with a description how to compile the Matlab interface on Windows
- compile_mex_all.m to be compatible with Windows
Add:
- env.m to set the environment varialbles in Windows using Matlab commands

giaf commented on interfaces/matlab_octave/env.m:
```diff
@@ -0,0 +1,15 @@
+setenv('BLASFEO_MAIN_FOLDER',[fileparts(fileparts(fileparts(pwd))),'/blasfeo']); % "$(pwd)/../../../blasfeo"
+setenv('MATLABPATH',[getenv('HPIPM_MAIN_FOLDER'),'/interfaces/matlab_octave/']); % $HPIPM_MAIN_FOLDER/interfaces/matlab_octave/
+setenv('OCTAVE_PATH',[getenv('HPIPM_MAIN_FOLDER'),'/interfaces/matlab_octave/']); %$HPIPM_MAIN_FOLDER/interfaces/matlab_octave/
+setenv('LD_LIBRARY_PATH',[getenv('HPIPM_MAIN_FOLDER'),'/lib/']); %$BLASFEO_MAIN_FOLDER/lib
```
This looks a bit different than in the sh file.
The aim is to append to (instead of overwrite) whatever is already in `LD_LIBRARY_PATH` the addresses of the folder where both HPIPM and BLASFEO libraries are.

giaf commented:
Thanks for the contribution, in general it looks good!
I would only ask you to fix the LD_LIBRARY_PATH as in the specific comment.

lenvm commented:
Thanks for looking at my pull request and your feedback.
I have updated it as follows:
- env.m now checks if the paths that are set are empty. If they are empty, it sets them. If they are not empty, they are appended.
- I updated the readme with instructions to copy the '.lib' files to the '/lib' folders

giaf commented:
Ok merged, thanks for the contribution!
